### PR TITLE
PIL-288 Add Connect tab

### PR DESCRIPTION
**Pull request by graszka22**

### CODE REVIEW

File: src/actions/walletConnectActions.js [hunk 1]
```diff
@@ -371,6 +372,11 @@ export const requestSessionAction = (uri: string) => {
   };
 };
 
+export const toggleWCPromoCardAction = (collapsed: boolean): WalletConnectTogglePromoCard => ({
+  type: TOGGLE_WALLET_CONNECT_PROMO_CARD,
+  collapsed,
```
Comment (jsidorenko) [inline src/actions/walletConnectActions.js]:
```suggestion
  payload: collapsed,
```

File: src/reducers/walletConnectReducer.js [hunk 2]
```diff
@@ -201,6 +210,12 @@ const walletConnectReducer = (
     case WALLETCONNECT_SESSION_RECEIVED:
       return { ...state, waitingForSession: false };
 
+    case TOGGLE_WALLET_CONNECT_PROMO_CARD:
+      return {
+        ...state,
+        promoCardCollapsed: action.collapsed,
```
Comment (jsidorenko) [inline src/reducers/walletConnectReducer.js]:
```suggestion
        promoCardCollapsed: action.payload.collapsed,
```

Comment (jsidorenko) [reply src/reducers/walletConnectReducer.js]: Or just:
```suggestion
        promoCardCollapsed: action.payload,
```
to follow the common approach

File: src/screens/WalletConnect/PromoCard.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+          source={promoImage}
+          resizeMode="cover"
+          style={{ width: '100%', aspectRatio: promoImageAspectRatio }}
+        />
+        <ContentContainer>
+          <BaseText center regular secondary>
+            Swap tokens on decentralized exchanges, lend &amp; borrow crypto, play games and more
+          </BaseText>
+          <ButtonsContainer>
+            <Button small primaryInverted title="Learn more" />
```
Comment (jsidorenko) [inline src/screens/WalletConnect/PromoCard.js]: Should we add some handlers for these buttons before going to production?

File: src/screens/WalletConnect/WalletConnectCallRequest.js [hunk 4]
```diff
@@ -121,105 +123,20 @@ class WalletConnectCallRequestScreen extends React.Component<Props, State> {
...
-    const {
-      fetchGasInfo,
-      session: { isOnline },
-    } = this.props;
-    if (prevProps.session.isOnline !== isOnline && isOnline) {
-      fetchGasInfo();
-    }
-  }
-
   handleBack = () => {
```
Comment (jsidorenko) [inline src/screens/WalletConnect/WalletConnectCallRequest.js]: Didn't find any references to this method. Could we remove it?

File: src/screens/WalletConnect/PromoCard.js (hunk 3 above)
Comment (graszka22) [reply src/screens/WalletConnect/PromoCard.js]: The "Learn more" section is not ready yet and should be done by the support team soon.
The "Explore" section might be assigned to @vyvil.

Comment (jsidorenko) [reply src/screens/WalletConnect/PromoCard.js]: Thanks! Dmitry just confirmed that we could merge that and the new screen will arrive shortly

**Comment by jsidorenko** [comment]
So happy to see how the Home screen's code gets smaller and smaller :) good job!

**Comment by jsidorenko** [comment]
All looks good from my side, but I would wait for @vyvil to give the final approval :)

File: src/screens/WalletConnect/withWCRequests.js [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    ...structuredSelector(state),
+    ...mapStateToProps(state),
+  });
+
+  const mapDispatchToProps = dispatch => ({
+    rejectCallRequest: (callId: number) => dispatch(rejectCallRequestAction(callId)),
+    fetchGasInfo: () => dispatch(fetchGasInfoAction()),
+  });
+
+  return withNavigation(connect(combinedMapStateToProps, mapDispatchToProps)(ComponentWithWCTransactions));
```
Comment (poocart) [inline src/screens/WalletConnect/withWCRequests.js]: This should be exported as default, but not appear within class return? 🤔

**Pull request merged**